### ISSUE TITLE
[Bug]: rename_collection() is not working as expected

### ISSUE THREAD

**Issue by Jordan-maker** [milvus-io/milvus#30743]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2): 2.3.6
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

When rename a created collection (let's say from 'coll' to 'renamed_coll') using a non-default database (let's say 'newdb'), the renamed collection is stored automatically in the default database (not in the current activated database), while the 'newdb' losses both the original and the renamed collection.

### Expected Behavior

when command utility.list_collections(), It should be printed ['renamed_coll'], but it appears in the default database, while the newdb finishes empty [].

### Steps To Reproduce

```markdown
from pymilvus import db, connections, utility
from pymilvus import FieldSchema, CollectionSchema, DataType, Collection

connections.connect(db_name="default", host="localhost", port="19530")

database = db.create_database(db_name="newdb")
db.using_database("newdb")

def create_collection(name, fields, description):
    schema = CollectionSchema(fields=fields, description=description, enable_dynamic_field=True)
    collection = Collection(name, schema, consistency_level="Strong")
    return collection

# Define fields for our collection
fields = [
    FieldSchema(name="book_id",
                dtype=DataType.INT64,
                is_primary=True,
                ),
    FieldSchema(name="book_name",
                dtype=DataType.VARCHAR,
                max_length=200,
                default_value="unknown",
                ),
    FieldSchema(name="book_intro",
                dtype=DataType.FLOAT_VECTOR,
                dim=2,
                description="introduction of the book",
                ),
    FieldSchema(name="work_count",
                dtype=DataType.INT64,
                default_value=9999,
                description="number of words that contains the book",
                )
]

collection = create_collection(name="coll",
                               fields=fields,
                               description="Collection contains relevant information about books")

utility.rename_collection("coll", "renamed_coll", timeout=None)
utility.list_collections(timeout=None)
```

**Comment by yanliang567** [comment]
I think this is a regression issue #27385 which was fixed before. @smellthemoon could you please double check why it reproduced on pymilvus 2.3.6?

/assign @smellthemoon  
/unassign

**Comment by xiaofan-luan** [comment]
@smellthemoon is that fixed?

**Comment by Jordan-maker** [comment]
I found that it is needed use the param new_db_name in the utility.rename_collection() function, even if the command  db.using_database("newdb") was previously run.

After run:
utility.rename_collection("coll", "renamed_coll", timeout=None, new_db_name="newdb")

The collection is renamed sucessfully in the newdb database (not in the default database).

I really thought that if the command db.using_database("newdb") is run previously, then it would not be needed to specify the new_db_name param.

**Comment by yanliang567** [comment]
I believe it is a bug. @smellthemoon could you please double check that?

**Comment by yanliang567** [comment]
verified fix on pymilvus 2.4.0
